### PR TITLE
Compute 2019-07-01 version

### PR DESCRIPTION
**Pull request by hyonholee**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by hyonholee** [comment]
@zikalino Could you check why model validation fails in this PR?  Thanks

**Comment by hyonholee** [comment]
@zikalino  the validation error says "Model definition 'DiskEncryptionSet' must have the properties 'name', 'id' and 'type' in its hierarchy and these properties must be marked as readonly."  But 'DiskEncryptionSet' has only id property which is set by a client in request body.

**Comment by hyonholee** [comment]
These validation failures are expected.

Potential new ARM errors

========================

R2020 - RequiredPropertiesMissingInResourceModel

Model definition 'DiskEncryptionSetParameters' must have the properties 'name', 'id' and 'type' in its hierarchy and these properties must be marked as readonly.

  at specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/compute.json:6879



R3016 - DefinitionsPropertiesNamesCamelCase

Property named: 'diskIOPSReadWrite', for definition: 'DataDisk' must follow camelCase style. Example: 'diskIopsReadWrite'.

  at specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/compute.json:7113



R3016 - DefinitionsPropertiesNamesCamelCase

Property named: 'diskIOPSReadWrite', for definition: 'VirtualMachineScaleSetDataDisk' must follow camelCase style. Example: 'diskIopsReadWrite'.

  at specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/compute.json:8420

**Comment by hyonholee** [comment]
@zikalino Hi, we finalized our Swagger spec from our side.  Could you review this PR?  The failed validation errors are expected.

**Comment by zikalino** [comment]
@hyonholee so we need to disable these validation errors

**Comment by qiaozha** [comment]
@hyonholee   For errors in LintDiff, if you'd like to suppress these validation errors, you may refer to specification/compute/resource-manager/readme.md of this repo and add them to the suppress part.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/compute.json [hunk 1]
```diff
@@ -2832,6 +2861,51 @@
...
+            "description": "The name of the virtual machine."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
```
Comment (KrisBash) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/compute.json]: Missing "default" response, which should be on all API responses. See[ here for an example. ](https://github.com/Azure/azure-rest-api-specs/blob/b51344bc3d2022a2ecacd5297a6ad1553418dd51/specification/batch/resource-manager/Microsoft.Batch/stable/2019-08-01/BatchManagement.json#L105)

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/compute.json [hunk 2]
```diff
@@ -4409,6 +4489,327 @@
...
+            "description": "Parameters supplied to the Create Virtual Machine Extension operation."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
```
Comment (KrisBash) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/compute.json]: These should all have "default" response too. Let's at least get the new APIs standarized in this fashion.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/disk.json [hunk 3]
```diff
@@ -0,0 +1,1842 @@
...
+            "name": "grantAccessData",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/GrantAccessData"
+            },
+            "description": "Access data object supplied in the body of the get disk access operation."
+          }
+        ],
+        "responses": {
```
Comment (KrisBash) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/disk.json]: Same comments on "default" throughout changed apis

**Review by KrisBash — CHANGES_REQUESTED** [review]
All swagger API specs should have a "default" response with a response body aligned to standard error contract. Can we at least add that in new/changed APIs?

**Comment by hyonholee** [comment]
@KrisBash   I added default responses to the new APIs.

**Review by KrisBash — APPROVED** [review]
LGTM thanks

**Review by zikalino — APPROVED** [review]

**Review by zikalino — APPROVED** [review]

**Pull request merged**